components/AddANote.vue

```diff
@@ -1,51 +1,61 @@
-<script setup>
-import { computed, ref, watch } from "vue";
+<script>
 import { usePackageState } from "../composables";
 
-const props = defineProps({
-  defaultNote: {
-    type: String,
-    default: "",
+export default {
+  name: "Add a Note",
+  props: {
+    name: {
+      type: String,
+      required: true,
+    },
   },
-  name: {
-    type: String,
-    required: true,
-  },
-});
-
-const noteRef = ref(props.defaultNote);
-const { addNote } = usePackageState();
+  setup() {
+    const { state, addNote } = usePackageState();
 
-const noteText = computed(() =>
-  noteRef.value ? "update note for this library" : "add a note for this library"
-);
-const notePlaceholder = "add note";
-const buttonText = ref("+");
-
-/*
-  Right now this just updates state —
-  we should make it actually save and give feedback
-*/
-const saveNote = () => {
-  buttonText.value = "...";
-  addNote({ pkg: props.name, note: noteRef.value });
-  setTimeout(() => (buttonText.value = "+"), 500);
+    return {
+      addNote,
+      state,
+    };
+  },
+  data() {
+    return {
+      tempFormValue: "",
```

Comment:
This is updated by the `updateText` method below. It however is not zero'd out automatically when the template re-renders with new props values. (The component is not unmounted, the rendered form updates only.)